### DISCUSSION TITLE
My branches have diveregd after a squash and merge. How do I fix that ?

### DISCUSSION THREAD

**Discussion by sebinbenjamin** [community/community#23249] [New to GitHub]
<p>I did a ‘Squash and merge’ to close a PR. Although all the changes are now present in both the develop and master, it still says the branches have diverged and there are still changes ahead and behind the default branch.</p>
<pre><code>This branch is 1 commit ahead, 28 commits behind develop.
</code></pre>
<p>How do I fix this? The 28 commits are the commits which were squashed during ‘squash and merge’.</p>
<p>I assumed that squash and merge would do a fast forward merge to prevent the divergence. I don’t want all the commits from develop to move to master, which is why I did a “squash and merge”, instead of rebasing.</p>

**Comment by airtower-luna** [comment]
<p>“Squash” in git terms means to combine a set of commits into one, that contains only the sum of changes, which is why you have only one new commit on the master branch.</p>
<p>If you want to keep the commits unchanged (which I generally consider a good idea!) and merge over the web GUI, you need to do a merge with a merge commit (see <a href="https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/about-pull-request-merges" rel="nofollow noopener">documentation on merging pull requests</a>). As far as I’m aware there is no way to do a fast-forward merge (without merge commit or rebase/squash) directly from a GitHub pull request, if that’s what you want you’ll have to merge locally and push to master.</p>

**Comment by sebinbenjamin** [comment]
<p>Ok. So I was reading the docs at <a href="https://help.github.com/en/github/administering-a-repository/about-merge-methods-on-github#squashing-your-merge-commits" rel="nofollow noopener">https://help.github.com/en/github/administering-a-repository/about-merge-methods-on-github#squashing-your-merge-commits</a> . It says "<em>Pull requests with squashed commits are merged using the <a href="https://git-scm.com/docs/git-merge#_fast_forward_merge" rel="nofollow noopener">fast-forward option</a>". </em> Doesn’t that mean that the branches will be even ?</p>

**Comment by airtower-luna** [comment, marked as answer]
<p>No. It only means the squashed commit will be added directly on top of to the target branch, without a merge commit.</p>

**Comment by sebinbenjamin** [comment]
<p>I totally get it now. Although I would say that the wording ‘squash and merge’ is a bit misleading.</p>

**Comment by KirranBain** [comment]
Can anyone explain what needs to be done after a squash and commit to overcome above mentioned situation ?
Considering I want to keep my dev commits history and not force rewrite it with that of main.

Thanks
